drivers/clock_control/CMakeLists.txt

```diff
@@ -48,3 +47,8 @@ endif()
 
 zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex_ll.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)
+
+if(CONFIG_CLOCK_CONTROL_RCAR_CPG_MSSR)
```

Comment:
perhaps we may create a new CMakeLists.txt in the renesas subfolder

Reply:
Yes, please create a separate CMakeList.txt file for each subdirectory. But is a new subdirectory really needed?

Reply:
The new sub directory could be a nice addition in my opinion if it was done for each platform.
In the current state, the renesas sub directory is indeed not really needed, at least from a functional point of view.

Reply:
I chose to move back drivers and headers into the common clock_control drivers directory.
If a new file tree is to be chosen for clarity, we would do it for all providers at the same time.